damage_window.py

```diff
@@ -0,0 +1,234 @@
+import cv2
+import numpy as np
+
+class Rectangle(object):
+    def __init__(self, x, y, w, h):
+        '''
+        x, y represents top left point
+        '''
+        self.x = x
+        self.y = y
+        self.w = w
+        self.h = h
+
+    def corners(self):
+        '''
+        returns top-left and bottom-right corners
+        '''
+        return [(self.x, self.y), (self.x + self.w, self.y + self.h)]
+
+    def offset(self, val):
+        self.x += val[0]
+        self.y += val[1]
+
+    def draw(self, img, color=(0,0,255), width=1):
+        cv2.rectangle(img, self.corners()[0], self.corners()[1], color, width)
+
+    def __repr__(self):
+        return '[({},{}),({},{})]'.format(self.x, self.y, self.w, self.h)
+
+
+class BoundingBoxObject(object):
+    def __init__(self, x, y, w, h, color=(0,0,255)):
+        cvtInt = lambda x : int(round(x))
+        self.bbox = Rectangle(cvtInt(x), cvtInt(y), cvtInt(w), cvtInt(h))
+        self.color = color
+
+    def draw(self, img):
+        self.bbox.draw(img, self.color)
+
+    def getImage(self, img):
+        corners = self.bbox.corners()
+        return img[corners[0][1]:corners[1][1],corners[0][0]:corners[1][0]]
+
+
+class DamageBar(BoundingBoxObject):
+    def __init__(self, x, y, w, h):
+        super(DamageBar, self).__init__(x, y, w, h, (255,0,0))
+
+        self.champion = BoundingBoxObject(
+            self.bbox.x + 2,
+            self.bbox.y,
+            self.bbox.h,
+            self.bbox.h,
+            color=(255,0,255)
+        )
+
+        self.dmgValue = BoundingBoxObject(
+            self.champion.bbox.x + self.champion.bbox.h,
+            self.bbox.y,
+            self.bbox.w - self.champion.bbox.w,
+            self.bbox.h / 2,
+            color=(0,0,255)
+        )
+
+        self.levelStars = BoundingBoxObject(
+            self.champion.bbox.x,
+            self.champion.bbox.y + 0.75 * self.champion.bbox.h,
+            self.champion.bbox.w,
+            self.champion.bbox.h * 0.25,
+            color=(0,255,255)
+        )
+
+    def draw(self, img):
+        super(DamageBar, self).draw(img)
+        self.champion.draw(img)
+        self.dmgValue.draw(img)
+        self.levelStars.draw(img)
+
+
+class DamageWindow(object):
+    def __init__(self, anchor, dim, padding, img=None):
+        self.damage_bars = DamageWindow.GenerateDamageBars(anchor, dim, padding, 10)
+
+        # this section is used to test if window is open
+        test_dim = (0.07 * dim[0], (dim[1] + padding) * 3 - padding)
+        self.open_test_section = BoundingBoxObject(anchor[0] - test_dim[0] - 2, anchor[1], test_dim[0], test_dim[1], (255,255,0))
```

Comment:
A couple of magic numbers. Is this test code necessary?

Reply:
Yes it is used to test if the window is open or not